clang/lib/3C/CtxSensAVarBounds.cpp

```diff
@@ -23,31 +22,30 @@
 // We will ignore array indexing.
 class StructAccessVisitor : public RecursiveASTVisitor<StructAccessVisitor> {
 public:
-  std::vector<std::string> StructAccessStr;
-  bool IsGlobal = false;
-
-  explicit StructAccessVisitor(ASTContext *Ctx) : C(Ctx) {
-    StructAccessStr.clear();
-  }
+  explicit StructAccessVisitor(ASTContext *Ctx) : C(Ctx), StructAccessStr() {}
   virtual ~StructAccessVisitor() { StructAccessStr.clear(); }
 
+  bool isGlobal() const { return IsGlobal; }
+
   void processVarDecl(VarDecl *VD) {
-    if (VD != nullptr &&
-        (VD->getType()->isPointerType() || VD->getType()->isStructureType())) {
+    assert("StructAccessVisitor visiting null VarDecl" && VD != nullptr);
+    if (VD->getType()->isPointerType() || VD->getType()->isStructureType()) {
+      // If VD is a ParmVarDecl isLocalVarDecl will return false
+      // FIXME: This method can be called multiple times, and each time it will
+      //        set the value of IsGlobal. This could be a problem if it is
+      //        called on a global and then a local variable. e.g.:
+      //        ( 0 ? global_struct_var : local_struct_var).array
```

Comment:
is this valid C syntax?

Reply:
Yes, it's valid. But i didn't find any real examples of it in any of our benchmarks.